### PR TITLE
UIDATIMP-721: Cover `<RecordTypesSelect>` component with unit tests

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose
Cover RecordTypesSelect component with unit tests

## Approach
Covered RecordTypesSelect component with unit tests using Jest + RTL

## Refs
https://issues.folio.org/browse/UIDATIMP-721

## Screenshots

image
image

### CODE REVIEW

**Review by IvanKryzhanovskyi — APPROVED** [review]

File: src/components/RecordTypesSelect/components/CompareRecordSelect.test.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const {
+      container,
+      getByText,
+    } = renderCompareRecordSelect(compareRecordSelectProps);
+    const compareContainer = container.querySelector('.borderLeft');
+
+    expect(getByText('Incoming records')).toBeDefined();
+    expect(getByText('Existing records')).toBeDefined();
+    expect(compareContainer).toBeDefined();
+  });
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/CompareRecordSelect.test.js:52]:
```suggestion
describe('when current language is LTR')
  it('diraction should be rendered correctly', () => {
    const {
      container,
      getByText,
    } = renderCompareRecordSelect(compareRecordSelectProps);
    const compareContainer = container.querySelector('.borderLeft');

    expect(getByText('Incoming records')).toBeDefined();
    expect(getByText('Existing records')).toBeDefined();
    expect(compareContainer).toBeDefined();
  });
```

File: src/components/RecordTypesSelect/components/CompareRecordSelect.test.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  it('should be rendered on the right side', () => {
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/CompareRecordSelect.test.js:54]: Same here

File: src/components/RecordTypesSelect/components/IncomingRecordMenu.test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  describe('when clicking on active option', () => {
+    it('appropriate record type should be chosen', () => {
+      const { getByText } = renderIncomingRecordMenu({ open: true });
+
+      fireEvent.click(getByText('Static value (submatch only)'));
+
+      expect(onClick.mock.calls[0][0].type).toEqual('STATIC_VALUE');
+    });
+  });
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/IncomingRecordMenu.test.js:61]: Add check that `MARC Holdings` and `MARC Authority` are greyed out (disabled), please
And would be nice to have such a check for existing records as well

File: src/components/RecordTypesSelect/components/RecordItem.test.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+          ...recordItemProps('holdings'),
+          isEditable: true,
+        });
+        const icon = container.querySelector('.icon');
+
+        expect(icon).toBeDefined();
+      });
+    });
+
+    describe('when initial button no icon', () => {
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/RecordItem.test.js:64]:
```suggestion
    describe('when initial button has no icon', () => {
```

File: src/components/RecordTypesSelect/components/RecordItem.test.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    describe('when clicking on disabled initial button', () => {
+      it('record should not be chosen', () => {
+        const { getByText } = renderRecordItem({
+          ...recordItemProps('holdings'),
+          isEditable: false,
+        });
+
+        fireEvent.click(getByText('Holdings'));
+
+        expect(onClick.mock.calls).toEqual([]);
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/RecordItem.test.js:98]:
```suggestion
        expect(onClick.mock.calls).toHaveLength(0);
```

File: src/components/RecordTypesSelect/RecordTypesSelect.test.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { fireEvent } from '@testing-library/react';
+
+import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';
+import '../../../test/jest/__mock__';
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/RecordTypesSelect.test.js:5]:
```suggestion
import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';

import '../../../test/jest/__mock__';
```

File: src/components/RecordTypesSelect/components/CompareRecordSelect.test.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';
+import '../../../../test/jest/__mock__';
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/CompareRecordSelect.test.js:4]:
```suggestion
import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';

import '../../../../test/jest/__mock__';
```

File: src/components/RecordTypesSelect/components/CompareRecordSelect.test.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      id={id}
+    />
+  );
+
+  return renderWithIntl(component, translationsProperties);
+};
+
+describe('CompareRecordSelect', () => {
+  describe('when current language is LTR', () => {
+    it('diraction should be rendered correctly', () => {
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/CompareRecordSelect.test.js:43]:
```suggestion
    it('direction should be rendered correctly', () => {
```

File: src/components/RecordTypesSelect/components/CompareRecordSelect.test.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+      const compareContainer = container.querySelector('.borderLeft');
+
+      expect(getByText('Incoming records')).toBeDefined();
+      expect(getByText('Existing records')).toBeDefined();
+      expect(compareContainer).toBeDefined();
+    });
+  });
+
+  describe('when current language is RLT', () => {
+    it('diraction should be rendered correctly', () => {
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/CompareRecordSelect.test.js:57]:
```suggestion
    it('direction should be rendered correctly', () => {
```

File: src/components/RecordTypesSelect/components/IncomingRecordMenu.test.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { noop } from 'lodash';
+import { fireEvent } from '@testing-library/react';
+
+import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';
+import '../../../../test/jest/__mock__';
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/IncomingRecordMenu.test.js:6]:
```suggestion
import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';

import '../../../../test/jest/__mock__';
```

File: src/components/RecordTypesSelect/components/IncomingRecordTrigger.test.js [hunk 11]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { noop } from 'lodash';
+
+import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';
+import '../../../../test/jest/__mock__';
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/IncomingRecordTrigger.test.js:5]:
```suggestion
import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';

import '../../../../test/jest/__mock__';
```

File: src/components/RecordTypesSelect/components/InitialRecordSelect.test.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { noop } from 'lodash';
+
+import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';
+import '../../../../test/jest/__mock__';
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/InitialRecordSelect.test.js:5]:
```suggestion
import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';

import '../../../../test/jest/__mock__';
```

File: src/components/RecordTypesSelect/components/RecordItem.test.js [hunk 13]
```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { fireEvent } from '@testing-library/react';
+
+import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';
+import '../../../../test/jest/__mock__';
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/RecordItem.test.js:5]:
```suggestion
import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';

import '../../../../test/jest/__mock__';
```

File: src/components/RecordTypesSelect/components/RecordSelect.test.js [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+    expect(getByText('Instance')).toBeDefined();
+    expect(getByText('Holdings')).toBeDefined();
+    expect(getByText('Item')).toBeDefined();
+    expect(getByText('MARC Holdings')).toBeDefined();
+    expect(getByText('MARC Bibliographic')).toBeDefined();
+    expect(getByText('Order & order line')).toBeDefined();
+    expect(getByText('Invoice')).toBeDefined();
+    expect(getByText('MARC Authority')).toBeDefined();
+  });
+  describe('when it renders on the right side', () => {
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/RecordSelect.test.js:35]:
```suggestion
  });
  
  describe('when it renders on the right side', () => {
```

File: src/components/RecordTypesSelect/components/RecordSelect.test.js [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+  });
+
+  describe('when it renders on the left side', () => {
+    it('container should be rendered with appropriate class', () => {
+      const { container } = renderRecordSelect({ isLocalLTR: true });
+
+      expect(container.querySelector('treeViewLTR')).toBeDefined();
+    });
+  });
+});
```
Comment (mariia-aloshyna) [inline src/components/RecordTypesSelect/components/RecordSelect.test.js:50]:
```suggestion
  describe('when current language is LTR', () => {
    it('direction should be rendered correctly', () => {
      const { container } = renderRecordSelect({ isLocalLTR: true });

      expect(container.querySelector('treeViewLTR')).toBeDefined();
    });
  });
});
```

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**